[MusicInfoLoader] Information for tracks within multi-track file not shown in playlist

78andyp opened the pull request:
## Description

The only metadata being loaded for tracks in a single multitrack file was from the CUE sheet in `PlayListM3U`.

Added code in `MusicInfoLoader` to load this based on the start offest - which is in the CUE file and the database - so, along with file name, can be used to identify the correct `idSong`.

Also it seems the play count was never being correctly updated for these tracks (it would always default to the first track), irrespective of how the song was played (ie through playlist or album).

## Motivation and context

Fixes #24247

## How has this been tested?

Reproduced from bug report and tested locally.

Will also request feedback from OP.

## What is the effect on users?

As above.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

fuzzard commented on xbmc/playlists/PlayListM3U.cpp:
```diff
@@ -178,7 +178,7 @@ bool CPlayListM3U::Load(const std::string& strFileName)
           newItem->SetEndOffset(iEndOffset);
           // Prevent load message from file and override offset set here
-          newItem->GetMusicInfoTag()->SetLoaded();
+          // newItem->GetMusicInfoTag()->SetLoaded(); - This prevents information being loaded for multi-song files later
```
so what behaviour does this change?

We cant just comment a line and leave it in. What was your reasoning that gave you pause to not just delete the line?

78andyp replied:
Only so that I can comment why I'd removed it. Will remove it.
As per the comment - if you mark the `MusicInfoTag` as loaded there (which is isn't really) then no information gets added later on.

the-black-eagle replied:
I'd question why (what) this prevents being loaded later (background loader maybe?).  However, I'd also suggest that CUE sheets are an obsolete method of loading songs, date from ~2000 and apart from a few niche cases (mka files - which I intend to address) are pretty much pointless these days.

I know we like to support legacy stuff as far as possible but I'd love to drop cue sheet support rather than extend or fix any issues with it.

78andyp replied:
I don't disagree but feel we should fix it and then depreciate it - then people can make the choice of uisng the last versoin that works or upgrading.

If i remember rightly it's because this:

```  if ((!pItem->HasMusicInfoTag() || !pItem->GetMusicInfoTag()->Loaded()) && pItem->IsAudio())```
(line 173 in MusicInfoLoader.cpp)

is false and the song info never getsl loaded.

78andyp committed:
[MusicInfoLoader] Information for tracks within multi-track file not shown in playlist

When tracks are added from a playlist the path refers to the music file on disc, not a musicdb:// path.

For multiple songs in a single WAV file, referenced from a CUE sheet this means song information is not loaded and playcount is not updated.

By finding the songs musicdb:// path this can be corrected.
